src/app/OperationalSessionSetup.cpp

```diff
@@ -291,6 +295,21 @@ CHIP_ERROR OperationalSessionSetup::EstablishConnection(const ReliableMessagePro
     mCASEClient = mClientPool->Allocate();
     ReturnErrorCodeIf(mCASEClient == nullptr, CHIP_ERROR_NO_MEMORY);
 
+    // TODO: Combine LargePayload flag with DNS-SD advertisements from peer
+    if (mSessionForLargePayload == TransportPayloadCapability::kLarge)
+    {
+#if INET_CONFIG_ENABLE_TCP_ENDPOINT
+        // Set the transport type for carrying large payloads
+        mDeviceAddress.SetTransportType(chip::Transport::Type::kTcp);
+#else
+        ChipLogError(Discovery, "Cannot setup session for large payload. Appropriate Transport unavailable");
```

Comment:
What about kBle, do we use connection oriented mode of BLE for Matter

Reply:
For operational communication BLE is not used. It is only used for Commissioning.